### PR TITLE
graph ql query and mutation

### PR DESCRIPTION
**Pull request by hossainahmed18**
Fixes #

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [x] Improvement
- [ ] New feature
- [ ] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## What was done

Serverless.yml file was changed for create apsync. necessary schema declared. Mapping templates describes how query and mutation will run

## How to test

- [x] Needs regression testing (add QA needed label)

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

## Reminder list

- Determine whether REST API Documentation should be updated
- Make an assessment on whether unit tests should be updated and/or added
- Ensure follow up tasks are linked to this PR

QA - if regression testing was requested:

- [ ] QA has verified

### CODE REVIEW

File: junayed-task/junayed-serverlesss-lambda/serverless.yml [hunk 1]
```diff
@@ -185,4 +187,34 @@ layers:
...
+        response: "getuserActivitiesByCustomerAndUser-response-mapping-template.txt"
+    schema: schema.graphql
+    serviceRole: "AppSyncServiceRole"
+    dataSources:
+      - type: AMAZON_DYNAMODB
+        name: userAvtivityDataSource
+        description: 'User Activity Table'
+        config:
+          tableName: '${env:TABLE_NAME}'
+          serviceRoleArn: { Fn::GetAtt: [DynamoDbTable, Arn] }
```
[the remark this answers is not in the document]
Comment (hossainahmed18) [reply junayed-task/junayed-serverlesss-lambda/serverless.yml]: this is automatically handling by  serverless-appsync-plugin

File: junayed-task/junayed-serverlesss-lambda/schema.graphql [hunk 2]
```diff
@@ -0,0 +1,32 @@
+type UserActivity {
+	pk: String!
+	sk: String!
+    gsi1pk: String
+    gsi1sk:String
```
[the remark this answers is not in the document]
Comment (hossainahmed18) [reply junayed-task/junayed-serverlesss-lambda/schema.graphql]: resolved with commit : https://github.com/CraftsmenLtd/coe-batch-q322/pull/317/commits/9f1b0eb124c5532e9b2af725358ce526203f6281

File: junayed-task/junayed-serverlesss-lambda/schema.graphql [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+type Mutation {
+	createUserAvtivity(input: UserActivityInput!): UserActivity
+}
+
+type Query {
+	getuserActivitiesByCustomerAndUser(customer: String!, user: String!): [UserActivity]
+	getuserActivitiesByUserAndDateRange(customer: String!, user: String!, startDate: String!, endDate:String!): [UserActivity]
+}
```
[the remark this answers is not in the document]
Comment (hossainahmed18) [reply junayed-task/junayed-serverlesss-lambda/schema.graphql]: Implemented query from lambda data source. Last time talking with you, you asked to create query only, that's why skipped mutation now.
commit: https://github.com/CraftsmenLtd/coe-batch-q322/pull/317/commits/9f1b0eb124c5532e9b2af725358ce526203f6281

File: junayed-task/junayed-serverlesss-lambda/schema.graphql [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+input UserActivityInput {
+	customer: String!
+	user: String!
+	uploadTime: String!
+	fileName: String!
+	fileSize: String!
+}
+
+input searchActivity {
```
Comment (rkzico21) [inline junayed-task/junayed-serverlesss-lambda/schema.graphql:21]: To make distinction between type and input it is better to define input name as SearchActivityInput. Also it seems you forgot to make it PascelCase.

File: junayed-task/junayed-serverlesss-lambda/schema.graphql [hunk 5]
```diff
@@ -0,0 +1,40 @@
+type UserActivity {
+	pk: String
+	sk: String
+    gsi1pk: String
+    gsi1sk:String
```
Comment (rkzico21) [inline junayed-task/junayed-serverlesss-lambda/schema.graphql:5]: Please remove those as these are not required

**Review by rkzico21 — CHANGES_REQUESTED** [review]
Please see the latest feedback. Apart from that change extension for all the vtl templates from .txt to .vtl

File: junayed-task/junayed-serverlesss-lambda/schema.graphql (hunk 5 above)
Comment (hossainahmed18) [reply junayed-task/junayed-serverlesss-lambda/schema.graphql]: resolved with commit : https://github.com/CraftsmenLtd/coe-batch-q322/pull/317/commits/8daf14765040c805144e4355389eeb511b5b6178

File: junayed-task/junayed-serverlesss-lambda/schema.graphql (hunk 4 above)
Comment (hossainahmed18) [reply junayed-task/junayed-serverlesss-lambda/schema.graphql]: resolved with commit : https://github.com/CraftsmenLtd/coe-batch-q322/pull/317/commits/8daf14765040c805144e4355389eeb511b5b6178

File: junayed-task/junayed-serverlesss-lambda/mapping-templates/createUserAvtivity-request-mapping-template.vtl [hunk 6]
```diff
@@ -0,0 +1,17 @@
+{
+    "version" : "2017-02-28",
+    "operation" : "PutItem",
+    "key" : {
+        "PK": $util.dynamodb.toDynamoDBJson($ctx.args.input.customer),
+        "SK": $util.dynamodb.toDynamoDBJson("$ctx.args.input.user-ctx.args.input.uploadedTime"),
```
Comment (rkzico21) [inline junayed-task/junayed-serverlesss-lambda/mapping-templates/createUserAvtivity-request-mapping-template.vtl:6]: Does this work as intended or it should be something like
"${ctx.args.input.user}-${ctx.args.input.uploadedTime}"

**Review by rkzico21 — CHANGES_REQUESTED** [review]
Just one minor query. Other than that everything looks fine.

File: junayed-task/junayed-serverlesss-lambda/mapping-templates/createUserAvtivity-request-mapping-template.vtl (hunk 6 above)
Comment (hossainahmed18) [reply junayed-task/junayed-serverlesss-lambda/mapping-templates/createUserAvtivity-request-mapping-template.vtl]: It works without brackets. But will not work without dollar. I accidently I removed the dollar while removing brackets. sorry for that mistake. commit: https://github.com/CraftsmenLtd/coe-batch-q322/pull/317/commits/b5ef627a83f2b5ae2087aea75178e8b19d0dcfad

**Review by rkzico21 — APPROVED** [review]
Everything looks okay. Good work.

**Pull request merged**